packages/chain-adapters/src/evm/types.ts

```diff
@@ -10,20 +10,20 @@ export type Account = {
 }
 
 export type BuildCustomTxInput = {
-  wallet: HDWallet
   accountNumber: number
   to: string
   data: string
   value: string
   gasLimit: string
-} & Fees
+} & Fees & { from?: string; wallet?: HDWallet }
```

Comment:
can wallet go away now that the switch chain side effect has been lifted out?

Reply:
and can we just smoosh `from` into the regular params as part of this refactor?

Reply:
I'm on this with cleanup in the common evm utils. Wasn't able to completely finish this evening, but will have sorted in the morning

Reply:
Actionned as part of the `buildCustomTx` / `buildCustomApiTx` split:

https://github.com/shapeshift/web/blob/ea408d795ad322bd2d824fe38a57c07e1c98fe3a/packages/chain-adapters/src/types.ts#L159